src/main/scala/ai/datahunters/md/util/ImageFeatureUtils.scala

```diff
@@ -0,0 +1,94 @@
+package ai.datahunters.md.util
+
+import java.awt.Color
+import java.awt.image.BufferedImage
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, File}
+
+import ai.datahunters.md.schema.BinaryInputSchemaConfig
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.transform.vision.image.opencv.OpenCVMat
+import com.intel.analytics.bigdl.transform.vision.image.{FeatureTransformer, ImageFeature, ImageFrame}
+import javax.imageio.ImageIO
+import org.apache.spark.sql.{DataFrame, Row}
+import org.slf4j.LoggerFactory
+
+object ImageFeatureUtils {
+
+  val UnknownLabel = "Unknown"
+  val JPGType = "jpg"
+  private val Logger = LoggerFactory.getLogger(ImageFeatureUtils.getClass)
+
+  def buildImageFrame(inputDF: DataFrame,
+                      labelsSize: Int,
+                      featureTransformer: FeatureTransformer,
+                      idCol: String = BinaryInputSchemaConfig.IDCol,
+                      fileDataCol: String = BinaryInputSchemaConfig.FileCol,
+                      basePathCol: String = BinaryInputSchemaConfig.BasePathCol,
+                      filePathCol: String = BinaryInputSchemaConfig.FilePathCol): ImageFrame = {
+    val inputRDD = inputDF
+      .rdd
+      .map(row => {
+        val originalImg: Array[Byte] = row.getAs(fileDataCol)
+        val id: String = row.getAs(idCol)
+        val basePath: String = row.getAs(basePathCol)
+        val filePath: String = row.getAs(filePathCol)
+        val label = Tensor(Array.fill(labelsSize)(0.0), Array(labelsSize))
+        try {
+          val bm = OpenCVMat.fromImageBytes(originalImg)
+          val finalBM = if (bm.channels() != 3) {
+            Logger.info(s"Image ${filePath} has not supported size of channel: ${bm.channels()}. Converting to: 3.")
+            val converted = convertToRGB(originalImg, filePath)
+            OpenCVMat.fromImageBytes(converted)
+          } else {
+            bm
+          }
+          val imgFeature = ImageFeature(originalImg, label, filePath)
+          imgFeature.update(ImageFeature.mat, finalBM)
+          imgFeature.update(idCol, id)
+          imgFeature.update(basePathCol, basePath)
+          Some(imgFeature)
+        } catch {
+          case e: Exception => {
+            Logger.warn(s"File ${filePath} could not be recognised as image (message: ${e}). Ignoring...")
+            None
+          }
+        }
+      }).filter(_.isDefined)
+      .map(_.get)
+    ImageFrame.rdd(inputRDD).transform(featureTransformer)
+  }
+
+  def imageFeatureToRow(labelsMapping: Map[Int, String],
+                        idKey: String = BinaryInputSchemaConfig.IDCol,
+                        basePathKey: String = BinaryInputSchemaConfig.BasePathCol,
+                        threshold: Float = 0.5f)(imageFeature: ImageFeature): Row = {
```

Comment:
If default threshold was passed via config I'm not sure if here is needed.

Reply:
@dawidsau It's jus default. I would leave it because in many cases 0.5 is the best option.